### PR TITLE
[Agent] Include beats in agent package

### PR DESCRIPTION
**Pull request by michalpristas**
When packaging an agent supported beats are packed into resulting package
New subfolder is created with artifacts of supported beats for specific os-arch combination (windows agent does not contain linux artifacts, 32 bit arch does not mix with 64...)

Fixes: elastic/beats#14915

### CODE REVIEW

**Comment by ph** [comment]
While testing this, I run the `mage package` task  in the `x-pack/agent` directory but I do not get the Metricbeat / Binaries binaries included in the agent linux zip. Are you expecting us to run `make snapshot` at the root?

File: dev-tools/packaging/packages.yml [hunk 1]
```diff
@@ -118,6 +131,12 @@ shared:
     <<: *common
     files:
       <<: *agent_binary_files
+      'beats/filebeat-{{.GOOS}}-{{.Platform.Arch}}.zip':
```
Comment (michalpristas) [inline dev-tools/packaging/packages.yml]: zip

File: dev-tools/packaging/packages.yml [hunk 2]
```diff
@@ -130,6 +149,12 @@ shared:
       uninstall-service-{{.BeatName}}.ps1:
         template: '{{ elastic_beats_dir }}/dev-tools/packaging/templates/windows/uninstall-service.ps1.tmpl'
         mode: 0755
+      'beats/filebeat-{{.GOOS}}-{{.Platform.Arch}}.zip':
```
Comment (michalpristas) [inline dev-tools/packaging/packages.yml]: version?

**Comment by ph** [comment]
@michalpristas it is ready for another round of review? I did a quick `mage package` and the required files appear to be present in the artifact?

**Comment by michalpristas** [comment]
@ph yeah it should be all working now, you need to check `build/distributions` inside `x-pack/agent` to verify content

**Comment by ph** [comment]
LGTM

**Pull request merged**